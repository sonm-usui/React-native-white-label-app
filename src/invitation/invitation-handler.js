import type {
  HandleInvitationsAction,
  InvitationPayload,
} from './type-invitation'
import { CONNECTION_INVITE_TYPES, HANDLE_INVITATION } from './type-invitation'
import { ensureVcxInitSuccess } from '../store/route-store'
import { call, put, select, takeLatest } from 'redux-saga/effects'
import {
  convertClaimOfferPushPayloadToAppClaimOffer,
  navigateToRoutePN,
} from '../push-notification/push-notification-store'
import {
  claimOfferRoute,
  homeDrawerRoute,
  homeRoute,
  invitationRoute,
  proofProposalRoute,
  proofRequestRoute,
  pushNotificationPermissionRoute,
  qrCodeScannerTabRoute,
} from '../common'
import { deepLinkProcessed } from '../deep-link/deep-link-store'
import { TYPE } from '../common/type-common'
import {
  getAllDid,
  getAllPublicDid,
  getClaimOffers,
  getProofRequests,
  getVerifiers,
} from '../store/store-selector'
import type { ClaimOfferPayload } from '../claim-offer/type-claim-offer'
import { CLAIM_OFFER_STATUS } from '../claim-offer/type-claim-offer'
import { claimOfferReceived } from '../claim-offer/claim-offer-store'
import type { ProofRequestPayload } from '../proof-request/type-proof-request'
import { PROOF_REQUEST_STATUS } from '../proof-request/type-proof-request'
import {
  presentationProposalSchema,
  validateOutofbandProofRequestQrCode,
} from '../proof-request/proof-request-qr-code-reader'
import { proofRequestReceived } from '../proof-request/proof-request-store'
import { schemaValidator } from '../services/schema-validator'
import type { VerifierData } from '../verifier/type-verifier'
import { proofProposalReceived } from '../verifier/verifier-store'
import type { OutOfBandNavigation } from '../qr-code/type-qr-code'
import { getPushNotificationAuthorizationStatus } from '../push-notification/components/push-notification-permission-screen'
import { Platform } from 'react-native'
import { usePushNotifications } from '../external-imports'
import {
  getExistingConnection,
  shouldSendRedirectMessage,
  getThreadId,
} from './invitation-helpers'
import { invitationReceived } from './invitation-store'
import { showSnackError } from '../store/config-store'
import { CONNECTION_ALREADY_EXIST } from '../connection-history/type-connection-history'
import {
  sendConnectionRedirect,
  sendConnectionReuse,
} from '../store/connections-store'
import Snackbar from 'react-native-snackbar'
import { convertAriesCredentialOfferToAppClaimOffer } from '../bridge/react-native-cxs/vcx-transformers'

function* checkExistingConnectionAndRedirect(
  invitation: InvitationPayload
): Generator<*, *, *> {
  const { publicDID, DID } = invitation.senderDetail

  const allPublicDid = yield select(getAllPublicDid)
  const allDid = yield select(getAllDid)

  const existingConnection = getExistingConnection(
    allPublicDid,
    allDid,
    publicDID,
    DID
  )

  if (existingConnection) {
    yield put(
      navigateToRoutePN(homeRoute, {
        screen: homeDrawerRoute,
        params: undefined,
      })
    )

    Snackbar.show({
      text: CONNECTION_ALREADY_EXIST,
      duration: Snackbar.LENGTH_LONG,
    })

    const sendRedirectMessage = shouldSendRedirectMessage(
      existingConnection,
      invitation,
      publicDID,
      DID
    )

    if (
      (invitation.type === CONNECTION_INVITE_TYPES.ARIES_V1_QR ||
        invitation.type === undefined) &&
      sendRedirectMessage
    ) {
      yield put(
        sendConnectionRedirect(invitation, {
          senderDID: existingConnection.senderDID,
          identifier: existingConnection.identifier,
        })
      )
    } else if (
      invitation.type === CONNECTION_INVITE_TYPES.ARIES_OUT_OF_BAND &&
      sendRedirectMessage
    ) {
      yield put(
        sendConnectionReuse(invitation.originalObject, {
          senderDID: existingConnection.senderDID,
        })
      )
    }
    return
  }

  yield put(invitationReceived({ payload: invitation }))

  const isAuthorized = yield call(getPushNotificationAuthorizationStatus)

  if (Platform.OS === 'ios' && usePushNotifications && !isAuthorized) {
    yield put(
      navigateToRoutePN(pushNotificationPermissionRoute, {
        senderDID: invitation.senderDID,
        navigatedFrom: homeRoute,
      })
    )
  } else {
    yield put(
      navigateToRoutePN(invitationRoute, {
        senderDID: invitation.senderDID,
        backRedirectRoute: homeRoute,
      })
    )
  }
}

function* handleOutOfBandNavigation({
  mainRoute,
  backRedirectRoute,
  uid,
  invitationPayload,
  senderName,
}: OutOfBandNavigation): Generator<*, *, *> {
  const isAuthorized = yield call(getPushNotificationAuthorizationStatus)

  if (Platform.OS === 'ios' && usePushNotifications && !isAuthorized) {
    yield put(
      navigateToRoutePN(pushNotificationPermissionRoute, {
        senderDID: invitationPayload.senderDID,
        navigatedFrom: qrCodeScannerTabRoute,
        intendedRoute: mainRoute,
        intendedPayload: {
          backRedirectRoute,
          uid,
          invitationPayload,
          senderName,
          hidden: true,
        },
      })
    )
  } else {
    yield put(
      navigateToRoutePN(mainRoute, {
        backRedirectRoute,
        uid,
        invitationPayload,
        senderName,
        hidden: true,
      })
    )
  }
}

function* onAriesOutOfBandInviteRead(
  invitation: InvitationPayload
): Generator<*, *, *> {
  const invite = invitation.originalObject
  if (!invitation) {
    yield put(navigateToRoutePN(homeRoute, {}))
    yield call(
      showSnackError,
      'QR code contains invalid formatted Aries Out-of-Band invitation.'
    )
    return
  }

  if (
    !invite.handshake_protocols?.length &&
    !invite['request~attach']?.length
  ) {
    // Invite: No `handshake_protocols` and `request~attach`
    // Action: show alert about invalid formatted invitation
    yield put(navigateToRoutePN(homeRoute, {}))
    yield call(
      showSnackError,
      'QR code contains invalid formatted Aries Out-of-Band invitation.'
    )
  } else if (
    invite.handshake_protocols?.length &&
    !invite['request~attach']?.length
  ) {
    // Invite: Has `handshake_protocols` but no `request~attach`
    // Action: Create a new connection or reuse existing.
    // UI: Show Connection invite
    yield call(checkExistingConnectionAndRedirect, invitation)
  } else if (invite['request~attach']?.length) {
    // Invite: Has `handshake_protocols` and has `request~attach`
    // Action:
    //  1. Create a new connection or reuse existing
    //  2. Rut protocol connected to attached action
    // UI: Show view related to attached action

    if (!invitation.attachedRequest || !invitation.attachedRequest[TYPE]) {
      yield put(navigateToRoutePN(homeRoute, {}))
      yield call(
        showSnackError,
        'QR code contains invalid formatted Aries Out-of-Band invitation.'
      )
      return
    }

    const uid = getThreadId(invitation.attachedRequest)
    const type_ = invitation.attachedRequest[TYPE]

    if (type_.endsWith('offer-credential')) {
      const claimOffer = yield call(convertAriesCredentialOfferToAppClaimOffer,
        invitation.attachedRequest
      )
      if (!claimOffer) {
        yield put(navigateToRoutePN(homeRoute, {}))
        yield call(
          showSnackError,
          'QR code contains invalid formatted Aries Out-of-Band invitation.'
        )
        return
      }

      const claimOffers = yield select(getClaimOffers)
      const existingCredential: ClaimOfferPayload = claimOffers[uid]

      if (
        existingCredential &&
        existingCredential.status === CLAIM_OFFER_STATUS.ACCEPTED
      ) {
        yield put(
          navigateToRoutePN(homeRoute, {
            screen: homeDrawerRoute,
            params: undefined,
          })
        )

        // we already have accepted that offer
        yield call(
          showSnackError,
          'The credential offer has already been accepted.'
        )
        return
      }

      yield put(
        claimOfferReceived(
          convertClaimOfferPushPayloadToAppClaimOffer(
            {
              ...claimOffer,
              remoteName: invitation.senderName,
              issuer_did: invitation.senderDID,
            },
            invitation.senderDID
          ),
          {
            uid,
            senderLogoUrl: invitation.senderLogoUrl,
            remotePairwiseDID: invitation.senderDID,
            hidden: true,
          }
        )
      )
      yield call(handleOutOfBandNavigation, {
        mainRoute: claimOfferRoute,
        backRedirectRoute: homeRoute,
        uid,
        invitationPayload: invitation,
        senderName: invitation.senderName,
      })
    } else if (type_.endsWith('request-presentation')) {
      const proofRequests = yield select(getProofRequests)
      const existingProofRequest: ProofRequestPayload = proofRequests[uid]

      if (
        existingProofRequest &&
        existingProofRequest.status === PROOF_REQUEST_STATUS.ACCEPTED
      ) {
        yield put(
          navigateToRoutePN(homeRoute, {
            screen: homeDrawerRoute,
            params: undefined,
          })
        )
        // we already have accepted that proof request
        yield call(
          showSnackError,
          'The proof request has already been accepted.'
        )
        return
      }

      const [outofbandProofError, outofbandProofRequest] = yield call(
        validateOutofbandProofRequestQrCode,
        invitation.attachedRequest,
        invitation.senderName
      )

      if (outofbandProofError || !outofbandProofRequest) {
        yield call(
          showSnackError,
          'QR code contains invalid formatted Aries Out-of-Band invitation.'
        )
        return
      }

      yield put(
        proofRequestReceived(outofbandProofRequest, {
          uid,
          senderLogoUrl: invitation.senderLogoUrl,
          remotePairwiseDID: invitation.senderDID,
          hidden: true,
        })
      )
      yield call(handleOutOfBandNavigation, {
        mainRoute: proofRequestRoute,
        backRedirectRoute: homeRoute,
        uid: uid,
        invitationPayload: invitation,
        senderName: invitation.senderName,
      })
    } else if (type_.endsWith('propose-presentation')) {
      if (
        !schemaValidator.validate(
          presentationProposalSchema,
          invitation.attachedRequest
        )
      ) {
        yield call(showSnackError, 'Invalid formatted Presentation Proposal.')
        return
      }

      const verifiers = yield select(getVerifiers)
      const existingVerifier: VerifierData = verifiers[uid]

      if (existingVerifier) {
        yield put(
          navigateToRoutePN(homeRoute, {
            screen: homeDrawerRoute,
            params: undefined,
          })
        )
        // we already have accepted that presentation proposal
        yield call(
          showSnackError,
          'The presentation proposal has already been accepted.'
        )
        return
      }

      yield put(
        proofProposalReceived(invitation.attachedRequest, {
          uid,
          senderLogoUrl: invitation.senderLogoUrl,
          senderName: invitation.senderName,
          remotePairwiseDID: invitation.senderDID,
          hidden: true,
        })
      )

      yield call(handleOutOfBandNavigation, {
        mainRoute: proofProposalRoute,
        backRedirectRoute: homeRoute,
        uid: uid,
        invitationPayload: invitation,
        senderName: invitation.senderName,
      })
    }
  } else {
    yield put(
      navigateToRoutePN(homeRoute, {
        screen: homeDrawerRoute,
        params: undefined,
      })
    )
    yield call(
      showSnackError,
      'QR code contains invalid formatted Aries Out-of-Band invitation.'
    )
  }
}

function* handleInvitationSaga(
  action: HandleInvitationsAction
): Generator<*, *, *> {
  const vcxResult = yield* ensureVcxInitSuccess()
  if (vcxResult && vcxResult.fail) {
    return
  }
  const invitation = action.invitation
  if (!invitation) {
    yield put(
      navigateToRoutePN(homeRoute, {
        screen: homeDrawerRoute,
        params: undefined,
      })
    )
  }

  // aries connection
  if (invitation.type === CONNECTION_INVITE_TYPES.ARIES_V1_QR) {
    yield call(checkExistingConnectionAndRedirect, invitation)
  } else if (invitation.type === CONNECTION_INVITE_TYPES.ARIES_OUT_OF_BAND) {
    // aries out-of-band connection
    yield call(onAriesOutOfBandInviteRead, invitation)
  } else {
    // proprietary connection
    yield call(checkExistingConnectionAndRedirect, invitation)
  }

  if (action.token) {
    yield put(deepLinkProcessed(action.token))
  }
}

export function* watchHandleInvitation(): any {
  yield takeLatest(HANDLE_INVITATION, handleInvitationSaga)
}
