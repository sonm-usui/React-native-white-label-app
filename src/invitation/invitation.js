// @flow
import React, { Component } from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import type { Store } from '../store/type-store'
import type { ResponseTypes } from '../components/request/type-request'
import type { InvitationProps, InvitationNavigation } from './type-invitation'

import { Container } from '../components'
import { homeDrawerRoute, homeRoute, invitationRoute } from '../common'
import { ResponseType } from '../components/request/type-request'
import { sendInvitationResponse, invitationRejected } from './invitation-store'
import { smsPendingInvitationSeen } from '../sms-pending-invitation/sms-pending-invitation-store'
import { SMSPendingInvitationStatus } from '../sms-pending-invitation/type-sms-pending-invitation'
import { Request } from '../components/request/request'
import { Platform } from 'react-native'
import { allowPushNotifications } from '../push-notification/push-notification-store'
import { usePushNotifications } from '../external-imports'

export class Invitation extends Component<InvitationProps, void> {
  render() {
    const { invitation } = this.props

    const {
      isValid,
      senderName,
      title,
      message,
      senderLogoUrl,
    } = isValidInvitation(this.props.invitation)

    if (!isValid) {
      return <Container />
    }

    return (
      <Container>
        <Request
          title={title}
          message={message}
          senderLogoUrl={senderLogoUrl}
          onAction={this.onAction}
          testID={'invitation'}
          invitationError={invitation ? invitation.error : undefined}
          senderName={senderName}
        />
      </Container>
    )
  }

  componentDidMount() {
    if (this.props.isSmsInvitationNotSeen) {
      this.props.smsPendingInvitationSeen(this.props.smsToken)
    }
  }

  hideModal = () => {
    const backRedirectRoute = this.props.route.params?.backRedirectRoute
    if (backRedirectRoute) {
      this.props.navigation.navigate(backRedirectRoute)
    } else {
      this.props.navigation.navigate(homeRoute, {
        screen: homeDrawerRoute,
        params: undefined,
      })
    }
  }

  onAction = (response: ResponseTypes) => {
    const { invitation } = this.props
    if (invitation) {
      const { payload } = invitation
      if (payload) {
        if (response === ResponseType.accepted) {
          if (Platform.OS === 'android' && usePushNotifications) {
            // for android, we don't need to show push permission screen
            // so we can safely register push notification on Android
            // For ios, we are already showing permission screen
            // and if user click on allow permission, then our saga would request
            // permission and update the FCM token
            this.props.allowPushNotifications()
          }
          this.props.sendInvitationResponse({
            response,
            senderDID: payload.senderDID,
          })
          this.props.navigation.navigate(homeRoute, {
            screen: homeDrawerRoute,
            params: undefined,
          })
        } else if (response === ResponseType.rejected) {
          this.props.invitationRejected(payload.senderDID)
          this.hideModal()
        }
      } else {
        this.hideModal()
      }
    }
  }
}

function isValidInvitation(
  invitation: $PropertyType<InvitationProps, 'invitation'>
) {
  let senderName = 'Anonymous'
  let title = 'Hi'
  let message = 'Anonymous wants to connect with you.'

  if (invitation && invitation.payload) {
    const { payload } = invitation
    senderName = payload.senderName || senderName
    message = `${senderName} wants to connect with you.`
    return {
      isValid: true,
      senderName,
      title,
      message,
      senderLogoUrl: payload.senderLogoUrl,
    }
  }

  return {
    isValid: false,
    senderName,
    title,
    message,
    senderLogoUrl: undefined,
  }
}

const mapStateToProps = (
  state: Store,
  { route: { params } }: InvitationNavigation
) => {
  const senderDID = params ? params.senderDID : ''
  const smsToken = params ? params.token : null
  const isSmsInvitationNotSeen =
    smsToken &&
    state.smsPendingInvitation[smsToken] &&
    state.smsPendingInvitation[smsToken].status !==
      SMSPendingInvitationStatus.SEEN

  return {
    invitation: state.invitation[senderDID],
    smsToken,
    isSmsInvitationNotSeen,
  }
}

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      sendInvitationResponse,
      invitationRejected,
      smsPendingInvitationSeen,
      allowPushNotifications,
    },
    dispatch
  )

export const invitationScreen = {
  routeName: invitationRoute,
  screen: connect(mapStateToProps, mapDispatchToProps)(Invitation),
}
