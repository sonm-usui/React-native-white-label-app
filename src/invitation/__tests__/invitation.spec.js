// @flow
import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { Invitation, invitationScreen } from '../invitation'

import { color } from '../../common/styles'
import { ResponseType } from '../../components/request/type-request'
import { homeRoute } from '../../common'
import { getNavigation, smsToken } from '../../../__mocks__/static-data'
import { Container } from '../../components'

describe('<Invitation />', () => {
  const ConnectedInvitation = invitationScreen.screen
  let store
  let dispatch
  let subscribe
  let navigation
  let route
  let sendInvitationResponse
  let smsPendingInvitationSeen
  let invitationRejected
  let instance
  let component
  let invitation
  let isSmsInvitationNotSeen
  let allowPushNotifications
  let props

  const firstInvitation = {
    requestId: 'requestId1',
    senderAgentKeyDelegationProof: {
      agentDID: 'EHNMj5FNjA3xBo8HUKQTZv',
      agentDelegatedKey: '8Esrno7vguYbwVJDa31aMCdWJTu2tC7tke8u2iKUUvgB',
      signature:
        'OXtzqX9LJJsBY/Mrhjff6b9L79Pdg8U4B7dI/3RR65BJr5jktAGMwpMLe1aQRQT0FlplZfNy8QtUd4KNPd+ZAg==',
    },
    senderDID: 'senderDID1',
    senderEndpoint: 'endpoint',
    senderLogoUrl: 'lu',
    senderName: 'sender1',
    senderVerificationKey: 'sVk',
    targetName: 'target name',
    senderDetail: {
      name: 'default',
      agentKeyDlgProof: {
        agentDID: 'EHNMj5FNjA3xBo8HUKQTZv',
        agentDelegatedKey: '8Esrno7vguYbwVJDa31aMCdWJTu2tC7tke8u2iKUUvgB',
        signature:
          'OXtzqX9LJJsBY/Mrhjff6b9L79Pdg8U4B7dI/3RR65BJr5jktAGMwpMLe1aQRQT0FlplZfNy8QtUd4KNPd+ZAg==',
      },
      DID: '3Bb6zCYMoGnMTho97HGDrn',
      logoUrl: 'http://www.evernym.com',
      verKey: '2BzhMgbLXbECPPYiLtSEd36Boxt1oquuGJeS1ofhPoEr',
    },
    senderAgencyDetail: {
      DID: 'BDSmVkzxRYGE4HKyMKxd1H',
      verKey: '6yUatReYWNSUfEtC2ABgRXmmLaxCyQqsjLwv2BomxsxD',
      endpoint: '52.38.32.107:80/agency/msg',
    },
  }

  const getState = () => ({
    invitation: {
      senderDID1: {
        error: null,
        isFetching: false,
        payload: firstInvitation,
        status: ResponseType.none,
      },
    },
    config: {
      showErrorAlerts: true,
    },
    connections: {
      connectionThemes: {
        default: {
          primary: `rgba(${color.actions.button.primary.rgba})`,
          secondary: `rgba(${color.actions.button.secondary.rgba})`,
        },
      },
    },
    user: {
      avatarName: undefined,
    },
    lock: {
      isTouchIdEnabled: false,
    },
    smsPendingInvitation: {},
    offline: {
      offline: false,
    },
  })

  beforeEach(() => {
    dispatch = jest.fn()
    subscribe = jest.fn()
    store = {
      getState,
      dispatch() {
        return dispatch
      },
      subscribe() {
        return subscribe
      },
    }
    navigation = {
      ...getNavigation({
        senderDID: firstInvitation.senderDID,
        smsToken,
      }),
    }
    route = {
      params: {
        senderDID: firstInvitation.senderDID,
        token: smsToken,
      },
    }
    const state = getState()
    invitation = state.invitation.senderDID1
    const showErrorAlerts = state.config.showErrorAlerts
    sendInvitationResponse = jest.fn()
    smsPendingInvitationSeen = jest.fn()
    invitationRejected = jest.fn()
    isSmsInvitationNotSeen = false
    allowPushNotifications = jest.fn()
    props = {
      invitation,
      showErrorAlerts,
      navigation,
      route,
      sendInvitationResponse,
      smsPendingInvitationSeen,
      invitationRejected,
      smsToken,
      isSmsInvitationNotSeen,
      allowPushNotifications,
    }
    component = renderer.create(
      <Provider store={store}>
        <Invitation {...props} />
      </Provider>
    )

    instance = component.root.findByType(Invitation).instance
  })
  afterEach(() => {
    sendInvitationResponse.mockReset()
    sendInvitationResponse.mockRestore()
    invitationRejected.mockReset()
    invitationRejected.mockRestore()
    smsPendingInvitationSeen.mockReset()
    smsPendingInvitationSeen.mockRestore()
    dispatch.mockReset()
    dispatch.mockRestore()
    subscribe.mockReset()
    subscribe.mockRestore()
  })

  it('should match snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ConnectedInvitation navigation={navigation} route={route} />
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should dispatch correct action for accepting invitation', () => {
    const response = ResponseType.accepted
    instance.onAction(response)

    expect(sendInvitationResponse).toHaveBeenCalledWith({
      response,
      senderDID: firstInvitation.senderDID,
    })
  })

  it('should reject invitation and redirect user to Home on Deny', () => {
    instance.onAction(ResponseType.rejected)
    expect(invitationRejected).toHaveBeenCalledWith(firstInvitation.senderDID)
    expect(navigation.navigate).toHaveBeenCalled()
    const redirectScreen = navigation.navigate.mock.calls[0][0]
    expect(redirectScreen).toBe(homeRoute)
  })

  it('should call smsPendingInvitationSeen action if isSmsInvitationNotSeen is true', () => {
    const invitationAfterAccept = {
      ...invitation,
      status: ResponseType.accepted,
    }
    component = renderer.create(
      <Provider store={store}>
        <Invitation
          {...props}
          invitation={invitationAfterAccept}
          isSmsInvitationNotSeen={true}
        />
      </Provider>
    )
    expect(smsPendingInvitationSeen).toHaveBeenCalledWith(smsToken)
  })

  it('should not call smsPendingInvitationSeen action if isSmsInvitationNotSeen is false', () => {
    const invitationAfterAccept = {
      ...invitation,
      status: ResponseType.accepted,
    }
    component = renderer.create(
      <Provider store={store}>
        <Invitation {...props} invitation={invitationAfterAccept} />
      </Provider>
    )
    expect(smsPendingInvitationSeen).not.toHaveBeenCalledWith(smsToken)
  })

  it('should return a empty container when invitation is null', () => {
    invitation = null
    navigation = {
      ...getNavigation({
        senderDID: null,
        smsToken: null,
      }),
    }
    const wrapper = renderer.create(
      <Provider store={store}>
        <Invitation
          {...props}
          invitation={invitation}
          navigation={navigation}
        />
      </Provider>
    )
    expect(wrapper.toJSON()).toMatchSnapshot()
    const instance = wrapper.root
    expect(instance.findByType(Container).children.length).toBe(1)
  })
})
