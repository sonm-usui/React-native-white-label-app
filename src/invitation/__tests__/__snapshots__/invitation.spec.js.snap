// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Invitation /> should match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      fifth={true}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "flex": 1,
        }
      }
    >
      <View
        hCenter={true}
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
          }
        }
        useNativeDriver={true}
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "flex-end",
            }
          }
        >
          <View>
            <View
              accessibilityLabel="invitation-text-container-message-title"
              accessible={true}
              center={true}
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                }
              }
              testID="invitation-text-container-message-title"
            >
              <Text
                accessibilityLabel="Hi"
                accessible={true}
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "fontSize": 19,
                    },
                    Object {
                      "fontFamily": "System",
                    },
                    Object {
                      "backgroundColor": "#ffffff",
                      "color": "#777777",
                    },
                    Object {
                      "fontWeight": "800",
                    },
                    Object {
                      "textAlign": "center",
                    },
                    null,
                    null,
                    null,
                    null,
                    null,
                    null,
                    null,
                    null,
                    null,
                    null,
                    null,
                    Object {
                      "lineHeight": 30,
                      "marginHorizontal": 30,
                      "marginVertical": 10,
                    },
                  ]
                }
                suppressHighlighting={true}
                testID="invitation-text-title"
              >
                Hi
              </Text>
              <Text
                accessibilityLabel="sender1 wants to connect with you."
                accessible={true}
                ellipsizeMode="tail"
                numberOfLines={2}
                onTextLayout={[Function]}
                style={
                  Object {
                    "color": "#505050",
                    "fontFamily": "System",
                    "fontSize": 13.399999999999999,
                    "fontWeight": "bold",
                    "margin": 10,
                    "textAlign": "center",
                  }
                }
                testID="invitation-text-message"
              >
                sender1 wants to connect with you.
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View>
            <View
              center={true}
              spaceBetween={true}
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                row={true}
                spaceBetween={true}
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "marginVertical": 40,
                  }
                }
                vCenter={true}
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  pointerEvents="box-only"
                >
                  <View
                    accessibilityLabel="invitation-avatars-invitee"
                    accessible={true}
                    imageShadow="imageShadow"
                    style={
                      Object {
                        "borderRadius": 38,
                        "elevation": 3,
                        "shadowColor": "rgba(0, 0, 0, 0.25)",
                        "shadowOffset": Object {
                          "height": 3,
                          "width": 0,
                        },
                        "shadowOpacity": 1,
                        "shadowRadius": 8,
                      }
                    }
                    testID="invitation-avatars-invitee"
                  >
                    <Image
                      accessibilityLabel="invitation-avatars-invitee-image"
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      source={
                        Object {
                          "default": "RelativeImageStub",
                        }
                      }
                      style={
                        Array [
                          Object {
                            "height": 76,
                            "width": 76,
                          },
                          Object {
                            "borderRadius": 38,
                          },
                        ]
                      }
                      testID="invitation-avatars-invitee-image"
                    />
                  </View>
                </View>
                <Image
                  source={
                    Object {
                      "default": "RelativeImageStub",
                    }
                  }
                  style={
                    Object {
                      "height": 8,
                      "marginHorizontal": 5,
                      "width": 60,
                    }
                  }
                />
                <View
                  accessibilityLabel="invitation-avatars-inviter"
                  accessible={true}
                  imageShadow="imageShadow"
                  style={
                    Object {
                      "borderRadius": 38,
                      "elevation": 3,
                      "shadowColor": "rgba(0, 0, 0, 0.25)",
                      "shadowOffset": Object {
                        "height": 3,
                        "width": 0,
                      },
                      "shadowOpacity": 1,
                      "shadowRadius": 8,
                    }
                  }
                  testID="invitation-avatars-inviter"
                >
                  <Image
                    accessibilityLabel="invitation-avatars-inviter-image"
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    source={
                      Object {
                        "uri": "lu",
                      }
                    }
                    style={
                      Array [
                        Object {
                          "height": 76,
                          "width": 76,
                        },
                        Object {
                          "borderRadius": 38,
                        },
                      ]
                    }
                    testID="invitation-avatars-inviter-image"
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "flexDirection": "column",
            "maxWidth": "100%",
            "padding": 14.357142857142858,
            "paddingBottom": 9.571428571428571,
            "width": "100%",
          },
          Object {
            "padding": 14.357142857142858,
            "paddingBottom": 28.714285714285715,
          },
        ]
      }
    >
      <View
        accessibilityLabel="invitation-accept"
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#86B93B",
            "borderRadius": 5,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": 15.314285714285713,
            "minWidth": "35%",
            "opacity": 1,
            "overflow": "hidden",
            "padding": 16.271428571428572,
            "paddingLeft": 9.571428571428571,
            "paddingRight": 9.571428571428571,
            "width": "100%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "fontSize": 19,
              },
              Object {
                "fontFamily": "System",
              },
              Object {
                "backgroundColor": "#333333",
                "color": "#ffffff",
              },
              Object {
                "fontWeight": "800",
              },
              Object {
                "textAlign": "center",
              },
              Object {
                "backgroundColor": "transparent",
              },
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
            ]
          }
          suppressHighlighting={true}
        >
          Connect
        </Text>
      </View>
      <View
        accessibilityLabel="invitation-deny"
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "stretch",
                "borderRadius": 5,
                "flexDirection": "row",
                "justifyContent": "center",
                "overflow": "hidden",
                "width": "100%",
              },
              Object {
                "alignItems": "center",
                "borderColor": "#CE0B24",
                "borderWidth": 1,
                "justifyContent": "center",
                "minWidth": "35%",
                "padding": 16.271428571428572,
                "paddingLeft": 9.571428571428571,
                "paddingRight": 9.571428571428571,
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "fontSize": 19,
                },
                Object {
                  "fontFamily": "System",
                },
                Object {
                  "backgroundColor": "#333333",
                  "color": "#ffffff",
                },
                Object {
                  "fontWeight": "600",
                },
                null,
                Object {
                  "backgroundColor": "transparent",
                },
                null,
                Object {
                  "color": "#d0021b",
                },
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
              ]
            }
            suppressHighlighting={true}
          >
            Deny
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`<Invitation /> should return a empty container when invitation is null 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;
